### DISCUSSION TITLE
Failed download https://streamyard.com/

### DISCUSSION THREAD

**Discussion by IkeMaldus** [aclap-dev/video-downloadhelper#612] [Help and Q&A]
### Your question

Version 9.0.1.18 (stable), target google, locale es-ES.

Version 9.0.1.18 (stable), target google, locale es-ES.

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://streamyard.com/watch/dD2addcdBiY5",
  "error": {
    "details": "SideDownload error: : [hls @ 000002339cffba80] Error when loading first segment 'https://onair-cdn.global.ssl.fastly.net/vod/UEIbUc7XDxwRMEnFbZouP/stream_720p_1716483501952_1_00.ts'\r\n[in#0 @ 000002339cffb8c0] Error opening input: Invalid data found when processing input\r\nError opening input file https://onair-cdn.global.ssl.fastly.net/vod/UEIbUc7XDxwRMEnFbZouP/vod_720p.m3u8?Authorization=3_20240523182102_1547a88a50ecd6bc0acfab40_0245fe16e6206fac35b9234b45ecb9a1bec67571_005_20240530182102_0026_dnld.\r\nError opening input files: Invalid data found when processing input\r\n",
    "downloadable_id": "downloadable_4439195879579539",
    "error": "coapp_failure",
    "id": "coapp_failure_1bda35c2-1d6e-4c46-ac2c-a7595ae4aeef"
  }
}

### Addon & CoApp version and details

Version 9.0.1.18 (stable), target google, locale es-ES.

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://streamyard.com/watch/dD2addcdBiY5",
  "error": {
    "details": "SideDownload error: : [hls @ 000002339cffba80] Error when loading first segment 'https://onair-cdn.global.ssl.fastly.net/vod/UEIbUc7XDxwRMEnFbZouP/stream_720p_1716483501952_1_00.ts'\r\n[in#0 @ 000002339cffb8c0] Error opening input: Invalid data found when processing input\r\nError opening input file https://onair-cdn.global.ssl.fastly.net/vod/UEIbUc7XDxwRMEnFbZouP/vod_720p.m3u8?Authorization=3_20240523182102_1547a88a50ecd6bc0acfab40_0245fe16e6206fac35b9234b45ecb9a1bec67571_005_20240530182102_0026_dnld.\r\nError opening input files: Invalid data found when processing input\r\n",
    "downloadable_id": "downloadable_4439195879579539",
    "error": "coapp_failure",
    "id": "coapp_failure_1bda35c2-1d6e-4c46-ac2c-a7595ae4aeef"
  }
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

I visited your web page & VDH had an empty menu.  So I launched playback in the player on the web page, then paused it after a few seconds.  That made VDH recognize 2 RawHLS items.  I tried to download one of them & got the same error you did.

So I went to Plan B.  I inspected the Network Monitor & found an HLS master manifest.  I did the usual & passed its URL through ffprobe.  I got an error saying 401 Unauthorized.  This appears to be the problem.  The web site has made its HLS master manifest inaccessible to outsiders like us.  You have to get access to the HLS master manifest in order to download the content but we are blocked.  This amounts to stealth DRM.  I believe we are simply out of luck here.  The web site is protecting its content against downloading.
